src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -147,6 +152,95 @@ public AppRuntimeSecretController(AuthorizationService authorizationService,
+    @PostMapping("/requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<String> addAppRequesterAppRuntimeSecret(@RequestHeader("Authorization") String authorization,
+                                                                  @PathVariable String requesterAddress,
+                                                                  @PathVariable String appAddress,
+                                                                  @PathVariable long secretIndex,
+                                                                  @RequestBody String secretValue) {
+        // TODO: remove following bloc once functioning has been validated
+        if (secretIndex > 0) {
+            log.error("Can't add more than a single app requester secret as of now." +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}]",
+                    requesterAddress, appAddress, secretIndex);
+            return ResponseEntity.badRequest().build();
+        }
+
+        if (secretIndex < 0) {
+            log.error("Negative index are forbidden for app requester secrets." +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}]",
+                    requesterAddress, appAddress, secretIndex);
+            return ResponseEntity.badRequest().build();
+        }
+
+        if (!SecretUtils.isSecretSizeValid(secretValue)) {
+            return ResponseEntity.status(HttpStatus.PAYLOAD_TOO_LARGE).build();
+        }
+
+        String challenge = authorizationService.getChallengeForSetAppRequesterAppRuntimeSecret(
+                requesterAddress,
+                appAddress,
+                secretIndex,
+                secretValue
+        );
+
+        if (!authorizationService.isSignedByHimself(challenge, authorization, requesterAddress)) {
+            log.error("Unauthorized to addAppRequesterAppRuntimeSecret" +
+                            " [requesterAddress:{}, appAddress:{}, expectedChallenge:{}]",
+                    requesterAddress, appAddress, challenge);
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+
+        if (teeTaskRuntimeSecretService.isSecretPresent(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                requesterAddress,
+                secretIndex)) {
+            log.error("Unauthorized to addAppRequesterAppRuntimeSecret: " +
+                            "secret already exists" +
+                            " [requesterAddress:{}, appAddress:{}, secretIndex:{}]",
+                    requesterAddress, appAddress, secretIndex);
+            return ResponseEntity.status(HttpStatus.CONFLICT).build(); // secret already exists
+        }
+
+        final Optional<TeeTaskRuntimeSecretCount> oAllowedSecretsCount =
+                teeTaskRuntimeSecretCountService.getAppRuntimeSecretCount(
```

Comment:
```suggestion
                teeTaskRuntimeSecretCountService.getRequesterAppRuntimeSecretCount(
```

Reply:
We pass the secret owner role so I don't think the method name should include `Requester`.

Reply:
Ok I see so we can discard my comment.
But we should rename it to something like getMaxAppRuntimeSecretCount

Reply:
Sure, that looks great! Thanks :)

Reply:
Done. I have renamed all "secret count" method into "max secret count"